src/bootstrap/builder.rs

```diff
@@ -849,7 +852,7 @@ impl<'a> Builder<'a> {
 
         // cfg(bootstrap): the flag was renamed from `-Zexternal-macro-backtrace`
         // to `-Zmacro-backtrace`, keep only the latter after beta promotion.
-        if stage == 0 {
+        if stage == 0 && cmd != "clippy" {
```

Comment:
Seems strange to gate this flag *only* on `stage == 0`, presumably it should be around the whole thing?